app/controllers/classified_listings_controller.rb

```diff
@@ -6,14 +6,15 @@ class ClassifiedListingsController < ApplicationController
   before_action :authenticate_user!, only: %i[edit update new dashboard]
 
   def index
-    @displayed_classified_listing = ClassifiedListing.find_by!(slug: params[:slug]) if params[:slug]
+    published_listings = ClassifiedListing.where(published: true)
+    @displayed_classified_listing = published_listings.find_by(slug: params[:slug]) if params[:slug]
```

Comment:
Should this still be `find_by!` with the exclamation point?

Reply:
Oops! Let me bring it back. Thanks, @rhymes!